File: assignments/assg3/190070006/code/q04.cpp
#include <map>
#include <set>
#include <list>
#include <cmath>
#include <ctime>
#include <deque>
#include <queue>
#include <stack>
#include <string>
#include <bitset>
#include <cstdio>
#include <limits>
#include <vector>
#include <climits>
#include <cstring>
#include <cstdlib>
#include <fstream>
#include <numeric>
#include <sstream>
#include <iostream>
#include <algorithm>
#include <unordered_map>




//Minor
//Akshay Iyer
//190070006
//h190070006 on hackerrank
//Email id on hackerrank- 190070006@iitb.ac.in

//I pledge on my honor that I have not given or received any unauthorized //assistance on this assignment or any previous task.



#include <bits/stdc++.h> 
using namespace std; 

int num;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
  
    int n;
    cin>>n;
    int n1;
    cin>>n1;
    vector <int> vect(n);

    

    for(int i=0;i<n;i++)
    {
        cin>>vect[i];
    }
    num=n;
    

    
    
    for(int j=0;j<n1;j++)
    {
    char c;
    cin>>c;
    if(c=='I')
    {
      int k;
      cin>>k;
      int k2;
      cin>>k2;
        k--;
      vect.insert(vect.begin() + k, k2); 
      num++;
    }
    if(c=='D')
    {
      int k;
      cin>>k;
      k--;
      vect.erase(vect.begin()+k);
        num--;
    }
    if(c=='R')
    {
      int k;
      cin>>k;
      if(k==1)
      {
      vect.insert(vect.begin(),vect[num-1]);
      vect.pop_back();
    }
        else if(k==-1){
            vect.push_back(vect[0]);
      vect.erase(vect.begin());
        }
    }
    if(c=='P')
    {
      int k;
      cin>>k;
        k--;
      cout<<vect[k];
      cout<<endl;
        
    }
    
    }
}

File: assignments/assg1/code/q04.py

#Minor
#Akshay Iyer
#190070006
#h190070006 on hackerrank
#Email id on hackerrank- 190070006@iitb.ac.in

#I pledge on my honor that I have not given or received any unauthorized #assistance on this assignment or any previous task.



def maxim(c, l, h, n,b):
    mid =l+((h - l)//2)
    if((mid==0 or c[mid-1]^b <= c[mid]^b) and (mid==n-1 or c[mid]^b >= c[mid+1]^b)): 
        return(mid) 
    elif(mid>0 and c[mid+1]^b >= c[mid]^b): 
        return(maxim(c,mid+1,h,n,b)) 
    else:
        return(maxim(c,l,mid-1,n,b)) 
      
n,q= [int(item) for item in input().split()]
n=int(n)
q=int(q)
#n=4
#q=3
#b=[2,4,5]
a=[]
b=[]
a = [int(item) for item in input().split()]
#a=[6,2,10,7]
xor1=0
for i in range(0, q): 
    k = int(input()) 
    xor1=xor1^k
    z=maxim(a,0,n-1,n,xor1)
    print(z+1,"\n")



